### PR TITLE
Adding approval of the revert request

### PR DESCRIPTION
**Pull request by ricardoamador**
## Description

Forgot to add auto-approval of the request. Since right now it will be validated correctly but we cannot merge due to the repository approval rules.

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: auto_submit/lib/service/approver_service.dart [hunk 1]
```diff
@@ -25,8 +25,15 @@ class ApproverService {
     final String? author = pullRequest.user!.login;
 
     if (!config.rollerAccounts.contains(author)) {
-      log.info('Auto-review ignored for $author');
-      return;
+      final List<String> labelNames = (pullRequest.labels as List<IssueLabel>)
+        .map<String>((IssueLabel labelMap) => labelMap.name)
+        .toList();
+      if (labelNames.contains(Config.kRevertLabel)) {
+        log.info('Auto-review approved for revert request.');
```
[the remark this answers is not in the document]
Comment (ricardoamador) [reply auto_submit/lib/service/approver_service.dart:31]: Not sure what other way there is to do this unless the author is not a user but a roller account.

[the remark this answers is not in the document]
Comment (ricardoamador) [reply auto_submit/lib/service/approver_service.dart:31]: I see your point. let me make a change.

File: auto_submit/lib/service/validation_service.dart [hunk 2]
```diff
@@ -197,8 +196,15 @@ class ValidationService {
     final GithubService githubService = await config.createGithubService(slug);
 
     if (revertValidationResult.result) {
-      ApproverService approverService = ApproverService(config);
-      await approverService.approve(messagePullRequest);
+      // Approve the pull request automatically as it has been validated.
```
Comment (godofredoc) [inline auto_submit/lib/service/validation_service.dart:199]: Maybe we can reuse the code on the approver service. Something like:

making `Future<void> approve(PullRequest pullRequest) async {` private removing the logic to check the rollers accounts, and then creating two wrapper methods: autoApproval that validates the autoroller accounts validation and then call the private approve and then revertApproval with the check for revert and then delegating to approve.

Comment (ricardoamador) [reply auto_submit/lib/service/validation_service.dart:199]: Okay that sounds good. I will look into that.

**Review by godofredoc — APPROVED** [review]

**Pull request merged**